### ISSUE TITLE
Video Directories

### ISSUE THREAD

**Issue by funkytupperware** [jcostom/rollerblades#31]
I have the docker image up and running.  However, I'm trying to get a directory full of prerolls to run.  The DailyPath updated Plex with "/preroll/Monday", "/preroll/Tuesday"...  but not the files within those directories.  If I set the "Months" in the JSON file to a directory, Plex is updated to have the preroll as the directory "/preroll" and not all the individual files located within the directory.  Since it doesnt seem as if Plex recognizes just a directory name, do you know how to get Rollerblades to list all the prerolls within a directory?
Thank you!

**Comment by jcostom** [comment]
I did a bit of testing. Sorry to say, you're probably not going to like my findings. :(

Here's my config:

```
$ cat prerolls.json
{
    "MONTHS": {
        "06": "/prerolls/months/june"
    },
    "HOLIDAYS": {
        "0401": "/prerolls/holiday/aprilfool",
        "1224": "/prerolls/holiday/christmas",
        "1225": "/prerolls/holiday/christmas",
        "1226": "/prerolls/holiday/christmas",
        "0214": "/prerolls/holiday/valentines",
        "1030": "/prerolls/holiday/halloween",
        "1031": "/prerolls/holiday/halloween",
        "1127": "/prerolls/holiday/thanksgiving",
        "1128": "/prerolls/holiday/thanksgiving"
    },
    "DAILYPATH": "/prerolls/daily"
}
```
I first went into the `/prerolls/daily` directory and created a directory called `tuesday`. I moved tuesday.mp4 and another video into that directory. I restarted rollerblades and see the following at the end of the output of `docker log rollerblades`:

```
[INFO] [02 Jul 2024 00:11:12 EDT] Change /prerolls/daily/monday.mp4 to /prerolls/daily/tuesday.mp4.
[INFO] [02 Jul 2024 01:11:13 EDT] Current preroll is: /prerolls/daily/tuesday.mp4.
[..bunch of the same message, every hour...]
[INFO] [02 Jul 2024 20:11:16 EDT] Current preroll is: /prerolls/daily/tuesday.mp4.
[INFO] [02 Jul 2024 20:31:26 EDT] Startup rollerblades.py/1.1.5.
[INFO] [02 Jul 2024 20:31:26 EDT] Loaded in prerolls data from /config/prerolls.json.
[INFO] [02 Jul 2024 20:31:26 EDT] Current preroll is: /prerolls/daily/tuesday.mp4.
[INFO] [02 Jul 2024 20:31:26 EDT] Change /prerolls/daily/tuesday.mp4 to /prerolls/daily/tuesday/tuesday.mp4;/prerolls/daily/tuesday/nfdark.mp4.
```

Next, I created a new "holiday", for July 2. I slipped this into the HOLIDAYS section:

`"0702": "/prerolls/testing"`

That directory contains 2 other video files. Restarted rollerblades, and this is at the end of the logs...

```
[INFO] [02 Jul 2024 20:36:03 EDT] Startup rollerblades.py/1.1.5.
[INFO] [02 Jul 2024 20:36:03 EDT] Loaded in prerolls data from /config/prerolls.json.
[INFO] [02 Jul 2024 20:36:03 EDT] Current preroll is: /prerolls/daily/tuesday/tuesday.mp4;/prerolls/daily/tuesday/nfdark.mp4.
[INFO] [02 Jul 2024 20:36:03 EDT] Change /prerolls/daily/tuesday/tuesday.mp4;/prerolls/daily/tuesday/nfdark.mp4 to /prerolls/testing/Plex4K.mp4;/prerolls/testing/preroll.mp4.
```

Bottom line? It's working as it should. Is your directory structure "correct"? Check out my config above - `DAILYPATH` is set to `/prerolls/daily`, and in that directory you find the days of the week. Case is important. Make those directories lower case.

I suspect that's your issue.

**Comment by funkytupperware** [comment]
Quite the opposite.  It looks likes my issue was that I never specified the volume that the preroll directory was located within my Docker Compose file.  Therefore, when I specified individual files in the JSON it worked fine, but was never able to search directories for all files.
Thanks for your help, everything is working perfectly now!

**Comment by jcostom** [comment]
Glad to hear it!
